supermarq-benchmarks/examples/qre-challenge/Lessons-Learned.md

```diff
@@ -0,0 +1,106 @@
+
+_Comparison and Implications for QRE_
+
+The choice of qubit architecture—synthetic versus neutral atom—has significant implications for QRE. Superconducting qubits often require careful consideration of connectivity due to their fixed physical layout, while neutral atoms offer more flexibility in qubit arrangement but face challenges in operation fidelity. A comprehensive QRE tool must account for these differences in architecture, adjusting resource estimations based on the type of qubit used, its coherence properties, and the specific operational constraints associated with each platform.
+
+### **Requirements**
+
+_Functional Requirements_
+
+- **Movement Analysis:** The tool must evaluate the impact of qubit movement within the architecture.
+
+- **Atom Reloading:** Consideration of the frequency and impact of atom reloading within quantum operations.
+
+- **Code Teleportation:** Ability to model and assess the implications of code teleportation on qubit resources.
+
+- **Connectivity:** Analysis of qubit connectivity and its effect on the overall architecture.
+
+- **Degree of Parallelization:** Estimation of the impact of parallel operations on qubit usage and efficiency.
+
+- **Custom Assumptions:** Flexibility to incorporate various architectural assumptions and parameters.
+
+- **Universal Gateset Configuration:** The tool must support the ability to implement a universal gateset without relying on magic state distillation, allowing for more versatile quantum operations.
+
+- **Flexible QEC Code Switching:** Users should be able to switch between different quantum error correction codes, such as 3D codes, subsystem codes, concatenated codes, and piece-able fault tolerance, providing a broader range of options for optimizing quantum systems.
+
+- **Seamless Integration:** The design should facilitate easy integration of new quantum error correction codes as they are developed, maintaining the tool’s relevance as the field advances.
+
+### **System Architecture**
+
+_High-Level Architecture_
+
+- **Input Module:** Accepts user-defined parameters for the quantum architecture, including movement constraints, reloading rates, and connectivity graphs.
+
+- **Analysis Engine:** Core computational module that performs the QRE calculations, considering all the defined parameters and assumptions.
+
+- **Output Module:** Generates reports and visualizations, providing insights into qubit resource usage and architectural trade-offs.
+
+- **User Interface:** A GUI that allows users to interact with the tool, input data, run simulations, and view results.
+
+_Detailed Architecture_
```

Comment:
it feels like there's something missing here - none of these subsystems actually deal with gateset implementation, or the code itself